### PR TITLE
Add GSSAPIAuthentication authentication class.

### PR DESCRIPTION
**Pull request by gkarg**
## Description

Add yet another Authentication method, via `requests-gssapi`. The provided class, `GSSAPIAuthentication` is complete copy-paste of `KerberosAuthentication`, but uses a much nicer library (`requests-gssapi`) underneath.

## Non-technical explanation

Added support for alternative method of kerberos authentication.

## Release notes

( ) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
(x) Release notes are required, with the following suggested text:

```markdown
* Add requests-gssapi based authentication via `GSSAPIAuthentication`. ({issue}`454`)
```

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -27,6 +27,7 @@
     readme = f.read()
 
 kerberos_require = ["requests_kerberos"]
+gssapi_require = ["requests_kerberos"]
```
Comment (hashhar) [inline setup.py:30]: this is same as `kerberos_require`. Why is this needed separately?

**Review by hashhar — COMMENTED** [review]
skimmed, Kerberos needs more attention so I'll revisit this.

File: setup.py (hunk 1 above)
Comment (gkarg) [reply setup.py]: spot on, this is a typo. fixed

File: setup.py [hunk 2]
```diff
@@ -27,6 +27,7 @@
     readme = f.read()
 
 kerberos_require = ["requests_kerberos"]
+gssapi_require = ["requests_gssapi"]
```
Comment (hovaesco) [inline setup.py:30]: add it to `all_require` maybe? just for sake of completeness

**Review by hovaesco — APPROVED** [review]

File: setup.py (hunk 2 above)
Comment (gkarg) [reply setup.py:30]: that's easy to add, but just to be clear: although those libraries do not conflict with each other, I would assume an end-user would want either one either another, but rarely both.

**Review by hashhar — APPROVED** [review]

**Comment by hashhar** [comment]
@cla-bot check
